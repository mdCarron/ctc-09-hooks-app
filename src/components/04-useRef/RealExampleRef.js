import React, { useState } from "react";
import MultipleCustomHooks from "../03-examples/MultipleCustomHooks";

const RealExampleRef = () => {
  const [show, setShow] = useState(false);

  return (
    <div className="container mt-5">
      <h1>Real Example Ref</h1>
      <hr />

      {show && <MultipleCustomHooks />}
      <button
        className="btn btn-outline-primary mt-5"
        onClick={() => {
          setShow(!show);
        }}
      >
        Show / Hide
      </button>
    </div>
  );
};

export default RealExampleRef;
