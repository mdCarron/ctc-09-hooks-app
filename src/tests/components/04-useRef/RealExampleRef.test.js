import React from "react";
import { shallow } from "enzyme";
import RealExampleRef from "../../../components/04-useRef/RealExampleRef";

describe("<RealExampleRef />", () => {
  const wrapper = shallow(<RealExampleRef />);

  test("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  test("should show <MultipleCustomHooks />", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
