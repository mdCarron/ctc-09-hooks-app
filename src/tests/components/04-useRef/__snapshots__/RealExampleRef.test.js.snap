// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RealExampleRef /> should match snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-5"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</div>
`;
